src/test/kotlin/uk/gov/justice/digital/hmpps/hmppselectronicmonitoringdataplatformapi/controller/LocationControllerTest.kt

```diff
@@ -532,4 +542,214 @@ class LocationControllerTest {
+
+    val expected = EmApiError("Duration should be from 1 to 24", HttpStatus.BAD_REQUEST)
+    val result = assertThrows<EmApiError> {
+      LocationController(locationService).aggregateLocationsByDeviceIdAndTimeFrameAndDuration(
+        deviceId,
+        startDate,
+        endDate,
+        duration,
+      )
+    }
+
+    confirmException(result, expected)
+    verify(locationService, times(0)).aggregateLocationsByDeviceIdAndTimeFrameAndDuration(
+      any<String>(),
+      any<Date>(),
+      any<Date>(),
+      any<Int>(),
+    )
+  }
+
+  @Test
+  fun `aggregateLocationsByDeviceIdAndTimeFrameAndDuration should return empty list when receive valid data but no data exist`() {
+    val deviceId = "3fc55bb7-ba52-4854-be96-661f710328fc"
+    val startDate = "2000-10-31T01:30:07.000-00:00"
+    val endDate = "2000-10-31T01:30:10.000-00:00"
+    val duration = 2
+
+    val expected = ResponseEntity(LocationAggregationResponse(message = "No data found"), HttpStatus.OK)
+    val result = LocationController(locationService).aggregateLocationsByDeviceIdAndTimeFrameAndDuration(
+      deviceId,
+      startDate,
+      endDate,
+      duration,
+    )
+
+    Assertions.assertThat(result.body?.locations).isEqualTo(expected.body?.locations)
+    confirmNoErrorForAggregation(result, expected)
+  }
+
+  @Test
+  fun `aggregateLocationsByDeviceIdAndTimeFrameAndDuration should return a list of data when insert valid data and data exists`() {
+    val deviceId = "3fc55bb7-ba52-4854-be96-661f710328fc"
+    val startDate = "2000-10-31T01:30:07.000-00:00"
+    val endDate = "2000-10-31T01:35:00.000-00:00"
+    val duration = 8
+
+    val datetime: Date = DateConverter().convertFromStringToDate("2000-10-31T00:00:00.000-00:00")
+
+    val locationDataList: List<LocationAggregation> = listOf(LocationAggregationDemo(25.0, 20.0, datetime))
+
+    val start: Date = DateConverter().convertFromStringToDate(startDate)
+    val end: Date = DateConverter().convertFromStringToDate(endDate)
+
+    Mockito.`when`(locationService.aggregateLocationsByDeviceIdAndTimeFrameAndDuration(deviceId, start, end, duration))
+      .thenReturn(locationDataList)
+
+    val expected: ResponseEntity<LocationAggregationResponse> =
+      ResponseEntity(LocationAggregationResponse(locationDataList), HttpStatus.OK)
+    val result = LocationController(locationService).aggregateLocationsByDeviceIdAndTimeFrameAndDuration(
+      deviceId,
+      startDate,
+      endDate,
+      duration,
+    )
+
+    Assertions.assertThat(result.body?.locations).isEqualTo(expected.body?.locations)
+    confirmNoErrorForAggregation(result, expected)
+    verify(locationService, times(1)).aggregateLocationsByDeviceIdAndTimeFrameAndDuration(
+      any<String>(),
+      any<Date>(),
+      any<Date>(),
+      any<Int>(),
+    )
+  }
+
+  @Test
+  fun `aggregateLocationsByDeviceIdAndTimeFrameAndDuration should return bad request when it does not receive valid IsoDateTime for startDate`() {
```

Comment:
Does this do something more than the test on line 572? They both seem to do about the same thing.